v2/snmp/manager.go

```diff
@@ -29,32 +35,181 @@ type Response struct {
 // If the function returns an error, the walk will be terminated.
 type Walker func(pdu *PDU) error
 
-// PDU defines ...
+// PDU defines an SNMP PDU.
 type PDU struct {
-	// TBD
+	RequestID   int32
+	Error       int
+	ErrorIndex  int
```

Comment:
Comment explaining ErrorIndex id's the variable in the varbind that caused the problem

Reply:
Will add appropriate comments as part of struct refactoring (in-progress).